### PR TITLE
Complete VIIRS_GRAN reader for NOAA CLASS - VIIRS EDR files

### PR DESCRIPTION
**Pull request by dhueholt**
After the Git fiasco I caused at #808 it was necessary to start from a new branch. This new branch is forked from the Pytroll Satpy master.

This pull request updates the VIIRS_GRAN reader originally created by @bbakernoaa. The reader now dynamically generates variables (based on the method in the TROPOMI reader by @tommyjasmin), requiring only the latitude/longitude to be described in YAML. The reader, formerly a copy of viirs_l1b, is changed to be fully compatible with VIIRS EDR files. All files are flake8 compatible. Appropriate unit tests are added to the test suite. These unit tests have been updated to fix (with any good luck) the issues pointed out by @djhoese in his comments on 808. 

This reader works on all types of VIIRS EDR files obtainable from NOAA CLASS, except for VIIRS Active Fires which are formatted very differently. The pre-existing ``viirs_edr_active_fires`` reader is designed to deal with those.

 - [ ] Closes #424 and #808
 - [ ] Tests reworked
 - [ ] Tests passed
 - [ ] Passes ``flake8`` on changed files
 - [ ] Fully documented
 - [ ] Documentation and licensing formatted to Satpy standard
 - [ ] Added my name (and Barry Baker's) to `AUTHORS.md`

### CODE REVIEW

File: satpy/tests/reader_tests/__init__.py [hunk 1]
```diff
@@ -37,9 +38,8 @@
                                       test_safe_sar_l2_ocn, test_viirs_edr_active_fires,
                                       test_hdfeos_base, test_modis_l2,
                                       test_electrol_hrit, test_mersi2_l1b,
-                                      test_avhrr_l1b_gaclac, test_vaisala_gld360,
-                                      test_fci_l1c_fdhsi, test_tropomi_l2,
-                                      test_hsaf_grib)
+                                      test_avhrr_l1b_gaclac, test_fci_l1c_fdhsi,
+                                      test_viirs_gran)
```
Comment (djhoese) [inline satpy/tests/reader_tests/__init__.py]: You accidentally removed a lot of these readers from the tests (see diff on github). The `test_vaisala_gld360`, `test_tropomi_l2`, and `test_hsaf_grib` reader tests have been removed from the import and the test suite at the bottom.

Comment (dhueholt) [reply satpy/tests/reader_tests/__init__.py]: Yikes! Added back now

File: satpy/etc/readers/viirs_gran.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+reader:
+  description: Generic JPSS VIIRS Products (Granule) Reader
+  name: viirs_gran
```
Comment (djhoese) [inline satpy/etc/readers/viirs_gran.yaml]: Not sure how I missed this, but the reader should probably be named `viirs_edr_gran` (the name in the YAML and the name of the YAML, python name change would be nice but optional). Thoughts?

Comment (dhueholt) [reply satpy/etc/readers/viirs_gran.yaml]: Good point, that's definitely a better name. I changed it in the newest commit.

**Comment by djhoese** [comment]
Wow I can't believe I forgot about this. So sorry @dhueholt and @tommyjasmin. If either of you can fix the merge conflicts that'd be great.

**Comment by tommyjasmin** [comment]
I guess that will be me Dave :-)
I will try and get the conflicts sorted today or tomorrow...

**Comment by djhoese** [comment]
Closing to avoid confusion. #951 is the newest version of this PR.

**Pull request closed without merging**